packages/e2e-tests/specs/editor/various/post-visibility.test.js

```diff
@@ -95,9 +95,7 @@ describe( 'Post visibility', () => {
 			'div[aria-label="Move forward to switch to the next month."]'
 		);
 		await (
-			await page.$x(
-				'//td[contains(concat(" ", @class, " "), " CalendarDay ")]/div[contains(concat(" ", @class, " "), " components-datetime__date__day ")][text() = "15"]'
-			)
+			await page.$x( '//td[@role="button"]/*[text() = "15"]' )
```

Comment:
Maybe we could make this a little bit more robust by adding, at the start of the selector, a parent element with `role="application"` and `aria-label="Calendar"` ?

![Screenshot 2022-02-17 at 22 11 40](https://user-images.githubusercontent.com/1083581/154571473-
f97da99f-b27d-446f-aefa-d280cf8e54d9.png)


We could probably apply this improvement also to the e2e tests added in #37491

Reply:
Ah, well spotted! It's done, and I'll put a separate PR together for the other tests you mentioned 😄